### PR TITLE
Fix issue #538

### LINKED ISSUE #538: error with deepcopy FSMSStorageProxy objects
An error occurs when trying to deepcopy an FSMSStorageProxy that contains a reference to an FSMContext object that contains a reference to a BaseStorage subclass object that contains a reference to asyncio.AbstractEventLoop object (for example, RethinkDBStorage or RedisStorage). I checked this on my PostgreSQL storage, but it also contains a reference to loop. Deepcopy starts sequentially copying all encapsulated objects to a new instance and an error occurs when trying to copy the loop.
In addition, there is no need to copy encapsulated objects for FSMSStorageProxy to work successfully. A copy of the dictionary (aka data) is enough for success.

* Operating System: Debian 10
* Python Version: last
* aiogram version: last
* aiohttp version: last
* uvloop version (if installed): last

### PR DESCRIPTION
**Pull request by p82o**
This fixes #538 issue and bring a bit of code beauty

Tested on i18n example with my PostreSQL Storage (to be pulled soon too)

### CODE REVIEW

File: aiogram/contrib/middlewares/fsm.py [hunk 1]
```diff
@@ -63,18 +57,18 @@ def state(self):
...
 
     async def save(self, force=False):
-        if self._copy != self or force:
+        if self._copy != dict(self) or force:
             await self.fsm_context.set_data(data=self)
         if self._is_dirty or force:
             await self.fsm_context.set_state(self.state)
         self._is_dirty = False
-        self._copy = copy.deepcopy(self)
+        self._copy = copy.deepcopy(dict(self))
```
Comment (uwinx) [inline aiogram/contrib/middlewares/fsm.py:65]: do a `dict(self)` once, lint a name to it and use it everywhere, where you use `self` as data

**Review by uwinx — CHANGES_REQUESTED** [review]

**Comment by uwinx** [comment]
We should deprecate `FSMProxy*` for safety reasons. (fsm-proxy is a known source of race conditions)

CC: @JrooTJunior

File: aiogram/contrib/middlewares/fsm.py (hunk 1 above)
Comment (p82o) [reply aiogram/contrib/middlewares/fsm.py:65]: self._copy is used to not to save changes when there isn't.
therefore dict(self) must be used every time whed we need to compare self as data with his copy.

**Comment by evgfilim1** [comment]
@p82o, a fix from #597 accidentally appeared here... :monocle_face: That's _not_ good
https://github.com/aiogram/aiogram/pull/539/commits/af1034e11fe9f299f6df95aee1a2ebacff5c4fe7

**Comment by p82o** [comment]
> @p82o, a fix from #597 accidentally appeared here... 🧐 That's _not_ good
> [af1034e](https://github.com/aiogram/aiogram/commit/af1034e11fe9f299f6df95aee1a2ebacff5c4fe7)

my bad. delete `text_decorations.py` from this commit please

**Comment by p82o** [comment]
> @p82o, a fix from #597 accidentally appeared here... 🧐 That's _not_ good
> [af1034e](https://github.com/aiogram/aiogram/commit/af1034e11fe9f299f6df95aee1a2ebacff5c4fe7)

Can I close this pull request and open a new one with changes from [uwinx's comment](https://github.com/aiogram/aiogram/pull/539#discussion_r603864995)? Due to lack of experience with github, I unfortunately deleted the branch with this pull request and can't make these changes here.

**Comment by evgfilim1** [comment]
> Can I close this pull request and open a new one with changes from [uwinx's comment](https://github.com/aiogram/aiogram/pull/539#discussion_r603864995)? 

Yes, of course :)

**Pull request closed without merging**